### PR TITLE
Client Write Operation

### PR DESCRIPTION
**Pull request by adityarao97**

### CODE REVIEW

**Comment by nsalazar0429** [comment]
Client should not be a route. Instead, it should be a stand alone entity that only handles its own files. Ex: JAVA Application, Web Site, etc

In this case, it can be a JAVA app since we don't need GUI

File: .idea/uiDesigner.xml [hunk 1]
```diff
@@ -0,0 +1,124 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (nsalazar0429) [inline .idea/uiDesigner.xml:2]: File should not be commited

File: chunkmaster/src/main/java/com/example/chunkmaster/ChunkMasterController.java [hunk 2]
```diff
@@ -13,7 +13,7 @@
 public class ChunkMasterController {
 
     // Record classes for request and response objects
-    public record Chunk(String uuid, String data) {}
+    public record Chunk(String id, String content) {}
```
Comment (nsalazar0429) [inline chunkmaster/src/main/java/com/example/chunkmaster/ChunkMasterController.java:16]: Let's change id to use `ChunkMetadata` that contains id and fileName

File: client/src/main/java/com/example/client/entity/Chunk.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+public class Chunk {
+    String id;
+    String content;
+}
```
Comment (nsalazar0429) [inline client/src/main/java/com/example/client/entity/Chunk.java:13]: Let's use the same model

File: client/src/main/java/com/example/client/service/ClientProcessor.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            }
+        }
+
+        byte[] byteArray = new byte[byteList.size()];
+        for (int i = 0; i < byteList.size(); i++) {
+            byteArray[i] = byteList.get(i);
+        }
+
+        return byteArray;
+        //converting byteArray back to file format and saving it
```
Comment (nsalazar0429) [inline client/src/main/java/com/example/client/service/ClientProcessor.java:40]: remove commented code

File: client/src/main/java/com/example/client/service/ClientService.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+//            ResponseEntity<Map> mappingResponse = restTemplate.getForEntity(
+//                    "http://localhost:8080/chunkmaster/chunkServers",
+//                    Map.class
+//            );
+//            Map<String, List<String>> chunkServerMapping = mappingResponse.getBody();
+//
+//            for (String server : chunkServerMapping.get(chunk)) {
+//                restTemplate.postForEntity(server + "/chunkServer/storeChunk?chunkId=" + chunk, fileData, String.class);
+//            }
+//        }
```
Comment (nsalazar0429) [inline client/src/main/java/com/example/client/service/ClientService.java:84]: commented code

File: files/mainCheck/001e26c7-a83a-4533-b356-a59483be9dd8.txt [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
���Dj������
+ɭ��ķ�̒��M@���C{�;ߜ�w�z�;I��tL�@Z� �HH�/�U?L�0w
+�@L�ԏ/6Wn�v}7:e���􃝐�R)ZZ��҄5Md���K�4�IYI�U�#�i&��/-�4aN�4��:�VZ���	k��H�!Z���	k��_����4�J�o�qSiX<,
V�w~�ސN�S�.o@�`�O���D(Dzܚ~�l
�
�a��co���".~�=~����2%�d�i��8b���ӳ2��G���v�z}0���
+o�/��{�/����Vց�[)^~n�2e��%��u�M����|�%A�|�+� Wv�0��
ҩ���������>p�l�����x�΍��H��Z0�$���0�<�S>n���4r�`/{�{|MK��K��\YY�{)jO툕}��U�M'�=#b7Z��f�L�Ni���6Z�L�Yw2HZV������\|�����/񮇷�/�i{ݜ��
#7��9�nז
�y�w�f����[iǪU��u��)��c\I�5�!	r}P��>>�8&�ȐE�O�ȭ��
```
Comment (nsalazar0429) [inline files/mainCheck/001e26c7-a83a-4533-b356-a59483be9dd8.txt:6]: This should not be commited , right ?

**Review by nsalazar0429 — CHANGES_REQUESTED** [review]
Let's not merge unneccesary files

**Review by Phuoc-L — DISMISSED** [review]
looks good

**Review by karandesai-sjsu — APPROVED** [review]
Good

**Review by nsalazar0429 — APPROVED** [review]

**Review by Phuoc-L — APPROVED** [review]
good

**Review by karandesai-sjsu — APPROVED** [review]
done

**Pull request merged**